src/Api/Controllers/OrganizationUsersController.cs

```diff
@@ -260,6 +260,17 @@ public async Task Put(string orgId, string id, [FromBody] OrganizationUserUpdate
             throw new NotFoundException();
         }
 
+        var organization = await _organizationRepository.GetByIdAsync(orgGuidId);
+        if (organization == null)
+        {
+            throw new NotFoundException();
+        }
+
+        if (model.Type == OrganizationUserType.Custom && !organization.UseCustomPermissions)
+        {
+            throw new BadRequestException("To enable custom permissions the organization must be on an Enterprise plan.");
+        }
+
```

Comment:
I don't think the controller is the right place to validate this.

1. we'll always need to check this regardless of where the request is coming from, so it should be in the service layer so that we don't accidentally skip it somewhere (indeed I think you might be missing some endpoints at the moment, e.g. what about inviting users in bulk?)

2. It's not where other validation checks live, check out the invite/update methods in `OrganizationService` and in particular `ValidateOrgnaizationUserUpdatePermissions`.

Reply:
You're quite right, as a general rule all logic should be placed inside the services; controllers should stick to handling the input/output models.
Thanks for the hint, I've updated `ValidateOrganizationUserUpdatePermissions`, that looks to be used everywhere.